pyproject.toml

```diff
@@ -0,0 +1,138 @@
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+  "setuptools >= 64",
+  "setuptools_scm >= 7",
+]
+
+[project]
+name = "idaes-pse"
+description="IDAES Process Systems Engineering Framework"
+readme = { file = "README.md", content-type = "text/markdown" }
+maintainers = [
+{ name = "Keith Beattie", email = "ksbeattie@lbl.gov" }
+]
+license = { text = "BSD" }
+urls = { Homepage = "https://idaes.org" }
+classifiers=[
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: End Users/Desktop",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Natural Language :: English",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: Unix",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Scientific/Engineering :: Chemistry",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+keywords=["IDAES", "idaes-pse", "energy systems", "chemical engineering", "process modeling"]
+dynamic = ["version"]
+dependencies = [
+  "pyomo >= 6.9.3",
+  "pint >= 0.24.1", # required to use Pyomo units. Pint 0.24.1 needed for Python 3.9 support
+  "networkx", # required to use Pyomo network
+  "numpy>=1,<3", # pandas constraint added on 2023-08-30 b/c bug in v2.1  # see IDAES/idaes-pse#1253
```

Comment:
This comment I suspect is on the wrong line

Reply:
> This comment I suspect is on the wrong line

@mrmundt do you mean the comment on L43, i.e. where the `numpy` dependency is defined? If so, I think you might be right, as the comment refers to `pandas`. So that comment should be moved to the line where the `pandas` dependency is defined.

As for the Python version, AFAIK the classifiers are not enforced. From a cursory search (https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires), `requires-python` seems to be the way to define this in `pyproject.toml`